dotcom-rendering/src/components/ShareButton.importable.tsx

```diff
@@ -0,0 +1,233 @@
+		}
+	}
+`;
+
+const copiedButtonStyles = (sizeXSmall: boolean) => css`
+	color: ${themePalette('--share-button-copied')};
+	padding: ${sizeXSmall ? '0 4px;' : '0 10px;'} svg {
+		fill: ${palette.success[400]};
+	}
+`;
+
+const nativeShare = (sizeXSmall: boolean) => css`
+	min-width: ${sizeXSmall ? '79px' : '105px'};
+	max-width: ${sizeXSmall ? '79px' : '105px'};
+	border-color: ${themePalette('--share-button-border')};
+`;
+
+const liveBlogMobile = (isCopied: boolean) => css`
+	${until.desktop} {
+		color: ${palette.neutral[100]};
+		border-color: rgba(255, 255, 255, 0.4);
+		svg {
+			fill: ${isCopied ? palette.success[500] : palette.neutral[100]};
+		}
+		${!isCopied &&
+		`:hover {
+			color: ${themePalette('--share-button-liveblog-mobile')};
+			background-color: ${palette.neutral[100]};
+			svg {
+				fill: ${themePalette('--share-button-liveblog-mobile')};
+			}
+		}`}
+	}
+`;
+
+const getUrl = ({
+	pageId,
+	CMP,
+	blockId,
+}: {
+	pageId: string;
+	CMP?: string;
+	blockId?: string;
+}) => {
+	const searchParams = new URLSearchParams({});
+	if (CMP) searchParams.append('CMP', CMP); // Do we want to track this?
+	if (blockId) searchParams.append('page', `with:block-${blockId}`);
+
+	const blockHash = blockId ? `#block-${blockId}` : '';
+	const paramsString = searchParams.toString();
+	return new URL(
+		`${pageId}${paramsString ? '?' + paramsString : ''}${blockHash}`,
+		'https://www.theguardian.com/',
+	).href;
+};
+
+export const NativeShareButton = ({
+	onShare,
+	size,
+	isLiveBlogMeta,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogMeta: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+
+	return (
+		<Button
+			onClick={() => onShare()}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={<SvgShare />}
+			css={[
+				buttonStyles,
+				nativeShare(sizeXSmall),
+				isLiveBlogMeta && liveBlogMobile(false),
+			]}
+		>
+			Share
+		</Button>
+	);
+};
+
+export const CopyLinkButton = ({
+	onShare,
+	size,
+	isLiveBlogMeta,
+	isCopied,
+}: {
+	onShare: () => void;
+	size?: Size;
+	isLiveBlogMeta: boolean;
+	isCopied: boolean;
+}) => {
+	const sizeXSmall = size === 'xsmall';
+	return (
+		<Button
+			onClick={() => onShare()}
+			size={size}
+			type="button"
+			priority="tertiary"
+			iconSide="left"
+			icon={isCopied ? <SvgCheckmark /> : <LinkIcon />}
+			css={[
+				...(isCopied
+					? [copiedButtonStyles(sizeXSmall)]
+					: [buttonStyles]),
+				sharedButtonStyles(sizeXSmall),
+				isLiveBlogMeta && liveBlogMobile(isCopied),
+			]}
+		>
+			{isCopied ? 'Link copied' : 'Copy link'}
+		</Button>
+	);
+};
+
+export const ShareButton = ({
+	size = 'small',
+	pageId,
+	blockId,
+	webTitle,
+	format,
+	context,
+}: Props) => {
+	const [isCopied, setIsCopied] = useState(false);
+	const [isShareSupported, setIsShareSupported] = useState(false);
+
+	const isLiveBlogMeta =
+		(format.design === ArticleDesign.LiveBlog &&
+			context === 'ArticleMeta') ||
+		context === 'SubMeta';
```

Comment:
Will this return `true` in all cases when `context === 'SubMeta'`, regardless of whether on a liveblog? If we want the liveblog submeta specifically I think this would be?

```suggestion
		format.design === ArticleDesign.LiveBlog &&
			(context === 'ArticleMeta' || context === 'SubMeta');
```

But actually are we sure the submeta should be included here? The background of the submeta on a liveblog doesn't pick up a theme colour, unlike the meta. Should it instead be:

```suggestion
		format.design === ArticleDesign.LiveBlog &&
			context === 'ArticleMeta';
```

which results in the following change on a liveblog:

| Before | After |
|--------|--------|
| ![share-before] | ![share-after] | 

[share-before]: https://github.com/guardian/dotcom-rendering/assets/53781962/f9048f67-cb6b-42cb-a1cf-12df825f2fa1
[share-after]: https://github.com/guardian/dotcom-rendering/assets/53781962/3138715b-ff39-4cc4-80ac-f2906e9a90d7